### PR TITLE
DESENG-689: Add image widget

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: [🎟️ DESENG-689](https://citz-gdx.atlassian.net/browse/DESENG-689)

**Description of changes:**
- **Feature** Add image widget
  - Added a new "ImageWidget" widget type in the API
  - Image widgets can have a title, optional description, uploaded image, and optional alt text
  - Added image widget option to the engagement authoring wizard
  - Added image widget display for the engagement view page

**User Guide update ticket (if applicable):**
https://citz-gdx.atlassian.net/browse/DESENG-705

### CODE REVIEW

**Comment by Baelx** [comment]
Should we create a ticket for documentation in the user guide? Anything holding us back from that?

**Comment by NatSquared** [comment]
> Should we create a ticket for documentation in the user guide? Anything holding us back from that?

Created a ticket - https://citz-gdx.atlassian.net/browse/DESENG-705

File: met-web/src/services/widgetService/MapService/index.tsx [hunk 1]
```diff
@@ -56,11 +61,12 @@ interface PreviewShapefileRequest {
...
 export const previewShapeFile = async (data: PreviewShapefileRequest): Promise<GeoJSON> => {
     try {
+        const url = Endpoints.Maps.SHAPEFILE_PREVIEW;
         const formdata = new FormData();
         if (data.file) {
-            formdata.append('file', data.file);
+            formdata.append('file', data.file, data.file.name);
         }
-        const response = await http.PostRequest<GeoJSON>(Endpoints.Maps.SHAPEFILE_PREVIEW, formdata);
+        const response = await http.PostRequest<GeoJSON>(url, formdata, {}, { 'Content-type': 'multipart/form-data' });
```
Comment (Baelx) [inline met-web/src/services/widgetService/MapService/index.tsx:69]: Was the linter throwing errors here?

**Review by Baelx — CHANGES_REQUESTED** [review]
Thanks for this Nat! Also nice of you to keep adding support for the old engagement view. We may want to try to save ourselves some work and ask Steve if we need to keep doing that.

Before proceeding, could you please just let me know:
- If all form labels are properly associated with controls
- If the uploaded image gets deleted when a widget is deleted. We'll need to make sure we're doing that

**Comment by NatSquared** [comment]
>Before proceeding, could you please just let me know:
>    * If all form labels are properly associated with controls  

✅ All form controls now have labels associated where possible. 
>    * If the uploaded image gets deleted when a widget is deleted. We'll need to make sure we're doing that

☑️ [Ticket created](https://citz-gdx.atlassian.net/browse/DESENG-706) to follow up with deleting images from widgets, engagements, etc

File: met-web/src/components/engagement/form/EngagementWidgets/Image/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            setPreviewImage(files[0]);
+            setValue('image_url', files[0].name, { shouldValidate: true, shouldDirty: true, shouldTouch: true });
+            return;
+        }
+
+        setPreviewImage(null);
+        setValue('image_url', '', { shouldValidate: true, shouldDirty: true, shouldTouch: true });
+    };
+
+    const handleUploadImageFile = async () => {
```
[the remark this answers is not in the document]
Comment (NatSquared) [reply met-web/src/components/engagement/form/EngagementWidgets/Image/Form.tsx:70]: No; there is currently no mechanism for this in MET

File: met-web/src/services/widgetService/MapService/index.tsx (hunk 1 above)
Comment (NatSquared) [reply met-web/src/services/widgetService/MapService/index.tsx:69]: This was a small bugfix I implemented to fix an issue with file uploading on the Map widget that turned out not to be related to the Image widget the way I thought it was. I accidentally left it in this PR; should I remove it?

Comment (Baelx) [reply met-web/src/services/widgetService/MapService/index.tsx:69]: You don't need to review it but for small, unrelated fixes, the PR description is a great place to add those!

**Pull request merged**